tests/tensor_search/test__httprequests.py

```diff
@@ -86,5 +89,58 @@ def run():
             return True
         assert run()
         
+    def test_opensearch_search_retry(self):
+        mock_post = mock.MagicMock()
+        mock_get = mock.MagicMock()
+        mock_response = requests.Response()
+        mock_response.status_code = 500
+        error_message = """HTTPSConnectionPool(host='internal-abcdefghijk-123456789.us-east-1.elb.amazonaws.com', port=9200):
+Max retries exceeded with url: /my-test-index-1/_mapping (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)'))
+"""
+        mock_get.side_effect = requests.exceptions.ConnectionError(error_message)
+        mock_post.side_effect = requests.exceptions.ConnectionError(error_message)
+        mock_environ = {
+            EnvVars.MARQO_MAX_BACKEND_SEARCH_RETRY_ATTEMPTS: str(3),
+            EnvVars.MARQO_MAX_BACKEND_ADD_DOCS_RETRY_ATTEMPTS: str(3),
+            "MARQO_BEST_AVAILABLE_DEVICE": "cpu"
+        }
 
+        @mock.patch('requests.post', mock_post)
+        @mock.patch('requests.get', mock_get)
+        @mock.patch('marqo._httprequests.ALLOWED_OPERATIONS', {mock_post, mock_get, requests.put})
+        @mock.patch.dict(os.environ, {**os.environ, **mock_environ})
+        def run():
+            try:
+                res = tensor_search.add_documents(
+                    config=self.config, add_docs_params=AddDocsParams(
+                        index_name=self.index_name_1,
+                        docs=[{"some ": "doc"}], auto_refresh=True, device="cpu"
+                    )
+                )
+                raise AssertionError
+            except BackendCommunicationError as e:
```

Comment:
Does this test actually check that the retry happens 3 times? Perhaps we can use `call_args_list` of the mocked HTTP requests to count the retries and ensure it is correct.

Reply:
done